storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -191,21 +195,27 @@
    */
   @Override
   public void startMetrics() {
-    metricsSystem.createIntegerGauge(
-        TekuMetricCategory.STORAGE,
-        "memory_state_count",
-        "Number of beacon states held in the in-memory store",
-        this::countStates);
-    metricsSystem.createIntegerGauge(
-        TekuMetricCategory.STORAGE,
-        "memory_block_count",
-        "Number of beacon blocks held in the in-memory store",
-        this::countBlocks);
-    metricsSystem.createIntegerGauge(
-        TekuMetricCategory.STORAGE,
-        "memory_checkpoint_state_count",
-        "Number of checkpoint states held in the in-memory store",
-        this::countCheckpointStates);
+    stateCountGauge =
+        Optional.of(
+            SettableGauge.create(
+                metricsSystem,
+                TekuMetricCategory.STORAGE,
+                "memory_state_count",
+                "Number of beacon states held in the in-memory store"));
+    blockCountGauge =
+        Optional.of(
+            SettableGauge.create(
+                metricsSystem,
+                TekuMetricCategory.STORAGE,
+                "memory_block_count",
+                "Number of beacon blocks held in the in-memory store"));
+    checkpointCountGauge =
+        Optional.of(
+            SettableGauge.create(
+                metricsSystem,
+                TekuMetricCategory.STORAGE,
+                "memory_checkpoint_state_count",
+                "Number of checkpoint states held in the in-memory store"));
```

Comment:
The counter fields aren't volatile so we should take a `writeLock` while creating these. Otherwise when we come to update them we may wind up getting a partially initialised object because there's no guarantee of visibility across threads.

All the places we'd update it would also have the write lock already so that works out nicer than making the fields `volatile`.